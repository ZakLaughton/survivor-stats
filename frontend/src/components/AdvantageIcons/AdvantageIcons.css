/* Tooltip container */
.tooltip {
    position: relative;
    display: inline-block;
  }
  
  /* Tooltip text */
  .tooltip .tooltiptext {
    visibility: hidden;
    width: 120px;
    background-color: rgb(0, 0, 0, 0.8);
    color: #fff;
    text-align: center;
    padding: 5px 0;
    border-radius: 6px;
   
    /* Position the tooltip text - see examples below! */
    bottom: 100%;
    left: 50%; 
    margin-left: -60px; /* Use half of the width (120/2 = 60), to center the tooltip */
    position: absolute;
    z-index: 1;
    text-transform: capitalize;
    font-family: 'Londrina Solid', sans-serif;
    font-weight: 100;
  }
  
  /* Show the tooltip text when you mouse over the tooltip container */
  .tooltip:hover .tooltiptext {
    visibility: visible;
  }

  /* Add arrow below tooltip */
  .tooltip .tooltiptext::after {
    content: " ";
    position: absolute;
    top: 100%; /* At the bottom of the tooltip */
    left: 50%;
    margin-left: -5px;
    border-width: 5px;
    border-style: solid;
    border-color: black transparent transparent transparent;
  }