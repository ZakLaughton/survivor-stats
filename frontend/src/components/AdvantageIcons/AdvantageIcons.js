import React from 'react';
import './AdvantageIcons.css';

const AdvantageIcons = ({ castaway }) => {
  const iconLibrary = {
    'immunity idol': 'fas fa-shield-alt',
    'idol nullifier': 'fas fa-ban',
    'vote steal': 'fas fa-user-minus',
    'legacy advantage': 'fas fa-scroll',
    'extra vote': 'fas fa-plus-circle',
    'fake idol': 'fab fa-pagelines',
  };

  /**
   * Takes an advantage name and returns the HTML element to render the icon.
   * NOTE: All icons use fontawesome EXCEPT the fake idol, which uses a .png
   * @param {string} advantageName
   * @return {element}
   */
  const getAdvantageIconElement = (advantageName) => {
    if (advantageName === 'fake idol') {
      return (
        <div className="tooltip advantage">
          <img alt="Fake Immunity Idol" src={require('../../img/fake_immunity_idol.png')} />
          <span className="animated fadeIn tooltiptext">{advantageName}</span>
        </div>
      );
    }
    return (
      <i className={`tooltip advantage ${iconLibrary[advantageName]}`}>
        <span className="animated fadeIn tooltiptext">{advantageName}</span>
      </i>
    );
  };

  return (
    <div className="tribe-advantage-container">
      {castaway.advantages
        && castaway.advantages.map((advantage, index) => (
          <div key={index}>{getAdvantageIconElement(advantage.item)}</div>
        ))}
    </div>
  );
};

export default AdvantageIcons;
