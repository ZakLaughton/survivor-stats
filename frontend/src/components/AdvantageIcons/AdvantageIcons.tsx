import React, { FunctionComponent } from 'react';
import './AdvantageIcons.css';
import styled from 'styled-components';
import { Advantage } from '../../types';

export interface AdvantageIconsProps {
  advantages: Advantage[];
}

interface AdvantagePng {
  alt: string;
  fileName: string;
}

interface PngLibrary {[key: string]: AdvantagePng}
interface IconLibrary {[key: string]: string}

const AdvantageIcons: FunctionComponent<AdvantageIconsProps> = ({ advantages }) => {
  const iconLibrary: IconLibrary = {
    'immunity idol': `fas fa-shield-alt`,
    'idol nullifier': `fas fa-ban`,
    'vote steal': `fas fa-user-minus`,
    'legacy advantage': `fas fa-scroll`,
    'extra vote': `fas fa-plus-circle`,
    'fake idol': `fab fa-pagelines`,
    'multi-choice advantage': `fas fa-arrows-alt`,
  };

  const pngLibrary: PngLibrary = {
    'immunity idol': {
      alt: `Immunity Idol`,
      fileName: `immunity_idol.png`,
    },
    'fake immunity idol': {
      alt: `Fake Immunity Idol`,
      fileName: `fake_immunity_idol.png`,
    },
    'half immunity idol 1': {
      alt: `Half Immunity Idol`,
      fileName: `half_immunity_idol_1.png`,
    },
    'half immunity idol 2': {
      alt: `Half Immunity Idol`,
      fileName: `half_immunity_idol_2.png`,
    },
  };

  /**
   * Takes an advantage name and returns the HTML element to render the icon.
   * NOTE: All icons use fontawesome EXCEPT the fake idol, which uses a .png
   * @param {string} advantageName
   * @return {element}
   */
  const getAdvantageIconElement = (advantageName: string) => {
    if (pngLibrary[advantageName]) {
      const advantage = pngLibrary[advantageName];
      return (
        <div className='tooltip advantage'>
          <img alt={advantage.alt} src={require(`../../img/${advantage.fileName}`)} />
          <span className='animated fadeIn tooltip-text'>{advantage.alt}</span>
        </div>
      );
    }
    return (
      <StyledFontAwesome className={`tooltip advantage ${iconLibrary[advantageName]}`}>
        <span className='animated fadeIn tooltip-text'>{advantageName}</span>
      </StyledFontAwesome>
    );
  };

  return (
    <div className='tribe-advantage-container'>
      {advantages &&
        advantages.map((advantage, index) => (
          <div key={index}>{getAdvantageIconElement(advantage.item)}</div>
        ))}
    </div>
  );
};

const StyledFontAwesome = styled.i`
  text-shadow: -1px 0 black, 0 2px black, 2px 0 black, 0 -1px black;
`;

export default AdvantageIcons;
