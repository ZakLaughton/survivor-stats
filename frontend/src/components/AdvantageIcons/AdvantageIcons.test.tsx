/* eslint-env jest */
import React from 'react';
import AdvantageIcons, { AdvantageIconsProps } from './AdvantageIcons';

const mockProps: AdvantageIconsProps = {
  advantages: [{ item: 'immunity idol' }, { item: 'fake immunity idol' }],
};

it('renders without crashing', () => {
  expect(<AdvantageIcons {...mockProps} />).toMatchSnapshot();
});
