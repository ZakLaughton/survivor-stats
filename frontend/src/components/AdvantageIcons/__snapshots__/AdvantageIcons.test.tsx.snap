// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<AdvantageIcons
  advantages={
    Array [
      Object {
        "item": "immunity idol",
      },
      Object {
        "item": "fake immunity idol",
      },
    ]
  }
/>
`;
